torch/_torch_docs.py

```diff
@@ -10317,8 +10321,14 @@ def merge_dicts(*dicts):
 
 Returns a tensor filled with uninitialized data. The shape and strides of the tensor is
```

Comment:
In light of the above discussion we should carefully update the documentation of `empty_strided`, too.

Let's start with the first paragraph:

"Creates a tensor with the specified :attr:\`size\` and :attr:\`stride\` and filled with undefined data."

This modernizes the language, preserves this PR's fix, and doesn't include implementation details like how much memory is allocated.

I don't know if a note about about how much memory an implementation of this function could use is interesting to users. So maybe we could drop it?

Now let's talk about the warning for this function. This warning is intended to say "don't create an overlapped tensor", and we can do that more easily (following the above):

"If the constructed tensor is "overlapped" (with multiple indices referring to the same element in memory) its behavior is undefined."

We don't declare "overlapped" behaviors as having undefined behavior in a note (yet), but we probably should.

Reply:
I agree. The amount of allocated memory doesn't sound that useful here.